backend/sirius-components-diagrams-layout/src/main/java/org/eclipse/sirius/components/diagrams/layout/incremental/IncrementalLayoutEngine.java

```diff
@@ -82,7 +82,6 @@ public IncrementalLayoutEngine(NodeSizeProvider nodeSizeProvider) {
     }
 
     public void layout(Optional<IDiagramEvent> optionalDiagramElementEvent, DiagramLayoutData diagram, ISiriusWebLayoutConfigurator layoutConfigurator) {
-        this.nodePositionProvider.reset();
```

Comment:
Previously, the nodePositionProvider was reset every time we required the layout of a diagram.

The effect of this reset was exclusively to set to null a field, named `last`. This field was used to remember the last node whose position was computed by the nodePositionProvider. This node can not be affected by the rest of the layout. 
The provider requires this information when several siblings need to be positioned so that they do not overlap. 

This reset mechanism caused two main issues : 
- The first one is probably personal, but I found it _odd_ that the nodePositionProvider was not 'self contained'. In this state, the provider actually depended on the layout algorithm to decide the position of the nodes. In my opinion, this goes against the logic that this component is supposed to be the one providing the position of the nodes.
- The second one is what caused the bug [837](https://github.com/eclipse-sirius/sirius-components/issues/837). It is easier to explain why with an example.
`
Take a parent node and two children. They are all being created at the same time, so none of those nodes have a position. 
The node provider pins the first child. Then it uses its field "last" to position the second child. The second child becomes "last".
Now we are positioning the parent. Since "last" is not reset, the parent is positioned after child two. 
`

This is why I remove this reset mechanism, and let the provider find the last positioned sibling by itslef.